instance/instance.go

```diff
@@ -53,32 +54,57 @@ const (
 //GetList a list of all instances
 func (s *Store) GetList(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
-	stateFilterQuery := r.URL.Query().Get("state")
-	var stateFilterList []string
-	if stateFilterQuery != "" {
-		stateFilterList = strings.Split(stateFilterQuery, ",")
-		if err := models.ValidateStateFilter(stateFilterList); err != nil {
-			log.ErrorCtx(ctx, err, nil)
-			http.Error(w, err.Error(), http.StatusBadRequest)
-			return
-		}
-	}
+	data := log.Data{}
+	var stateFilterQuery string
+	var ap common.Params = nil
```

Comment:
Is there a need to set this to nil here?

Reply:
Yes. The state query may or may not be nil & we only want to record it if its not empty. I set this `nil` here so that audit params is either nil or a populated struct.

Reply:
Sorry I think I didn't make this clear, if we change `var ap common.Params = nil` to just `var ap common.Params`; ap would still be equal to nil and not an empty struct.